src/python/antares_xpansion/input_checker.py

```diff
@@ -292,50 +427,62 @@ def check_setting_option_type(option, value):
                  False or exits if the value has the wrong type
     """
 
-    options_types = {
-        "method": "string",
-        "uc_type": "string",
-        "master": "string",
-        "optimality_gap": "double",
-        "relative_gap": "double",
-        "cut_type": "string",
-        "week_selection": "string",
-        "max_iteration": "integer",
-        "relaxed_optimality_gap": "string",
-        "solver": "string",
-        "timelimit": "integer",
-        "yearly-weights": "string",
-        "additional-constraints": "string",
-        "log_level": "integer",
-    }
-    option_type = options_types.get(option)
-    if option_type is None:
+    if options_types_and_legal_values.get(option) is None:
         flushed_print(
             'check_setting_option_type: Illegal %s option in settings file.' % option)
-        sys.exit(1)
-    else:
-        if option_type == 'string':
+        raise NotHandledOption
+
+    option_type = options_types_and_legal_values.get(option)[0]
+
+    if option_type == type_float:
+        try:
+            float(value)
             return True
-        elif option_type == 'numeric':
-            return value.isnumeric()
-        elif option_type == 'double':
-            try:
-                float(value)
-                return True
-            except ValueError:
-                return False
-        elif option_type == 'integer':
-            if value in ["+Inf", "-Inf", "+infini", "-infini"]:
-                return True
+        except ValueError:
+            flushed_print(
+                'check_setting_option_type: Illegal %s option in type, numerical value is expected .' % option)
+            return False
+    elif option_type == type_int:
+        if value in ["+Inf", "-Inf", "+infini", "-infini"]:
```

Comment:
We do not want to do this for  `log_level` and we should not allow `-Inf` or `-infini` for the other parameters

Reply:
checking for loglevel is not problematic